src/pages/CourseDetails/CourseDetails.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {useEffect, useState} from 'react';
+import {useParams} from 'react-router-dom';
+import {getCourseDetailsById} from '../../services/courseService.ts';
+import {CourseProps} from '../../services/types.ts';
+import {SmallHeader} from "../../components/SmallHeader/SmallHeader.tsx";
+
+import './CourseDetails.css';
+import {GolfDetailsCard} from "../../components/GolfDetailsCard/GolfDetailsCard.tsx";
+import CourseCardSelector from "../../components/CourseCardSelector/CourseCardSelector.tsx";
+import {ReturnButton} from "../../components/Buttons/ReturnButton/ReturnButton.tsx";
+import {HolesCard} from "../../components/HolesCard/HolesCard.tsx";
+
+const CourseDetails: React.FC = () => {
+    const {courseId} = useParams<{ courseId: string }>();
+    const [courseDetails, setCourseDetails] = useState<CourseProps | null>(null);
+    const [error, setError] = useState<string | null>(null);
+
+    useEffect(() => {
```

Comment:
As discussed last time, you can better encapsulate your code into custom hooks such as 'useCourseDetails'.

think of the hierarchy this way:
useCourseDetails -> useAsync -> actual api call and error handling

the response format from your useAsync is very important in making your code clean.  It could return you
[error, loadingState, results] in one call.